### ISSUE TITLE
Macro functions that use enum values for input our output should use the enum name instead of English display name

### ISSUE THREAD

**Issue by Irarara** [RPTools/maptool#2595]
**Is your feature request related to a problem? Please describe.**
Related to #2583. Some macro functions that deal with certain properties (largely those on tokens) that are stored as enums return or take the English display names of the property value instead of the enum name (ex. `TOP_DOWN` vs `Top Down`). For the sake of consistency and because it's more localization friendly, it would be nice if everything used the enum names.

Examples (not all inclusive, still investigating):
* `setTokenShape()`/`getTokenShape`: enum name or English display name work for input, but they return the English display names (`Top Down`, `Circle`, `Square`, `Figure`).  These should return `TOP_DOWN`, `CIRCLE`, `SQUARE`, and `FIGURE`.
* `setTokenVBL()`: Uses the English display names for the `method` parameter (see the [wiki page](https://wiki.rptools.info/index.php/setTokenVBL)).  This should use the `JTS_SimplifyMethodType` names, `DOUGLAS_PEUCKER_SIMPLIFIER`, `TOPOLOGY_PRESERVING_SIMPLIFIER`, `VW_SIMPLIFIER`, and `NONE`.

This change does run the risk of breaking users' macros, though I will point out for both examples above the current wiki documentation regarding the values is either incorrect or incomplete. Also, for say, `getTokenShape()` it's possible someone out there is using it because they want to display the string somewhere (and the enum name would look kind of odd), but that's easy enough to clean up in a macro, or if it was something really in demand, it'd be better to add a parameter to the function that would return the localized name, or something along those lines.

Last thought, there's some places where either input is accepted. for instance `setTokenShape()` allows `top down` or `top_down`, and `setLayer()` allows `hidden` as a substitute for `gm`. My feeling is that since there isn't much harm in it this behavior should just be grandfathered in, but I dunno if anyone feels like it'd be better to be strict about it.

**Describe the solution you'd like**
Change all applicable macro functions to use the enum names.

**Describe alternatives you've considered**
The alternative would be to maintain the current behavior. It wouldn't be too hard to localize the displayed text and do this, but it would maintain the current inconsistency present with some functions and macro functions ideally shouldn't be using regular English text for values.

**Comment by Phergus** [comment]
I'm for changing these to expect and return the enum names while preserving backwards-compatibility for the `set*()` functions that currently accept the alternatives.

I'm not real worried about getting display strings as we do have `lower()`, `upper()` and `capitalize()` functions to help with that if needed.

**Comment by Irarara** [comment]
> I'm not real worried about getting display strings as we do have lower(), upper() and capitalize() functions to help with that if needed.

Yep, yep. Though, thinking about it did make me think it'd be kind of nice to be able to fetch the localized names, but that's a wholly separate issue. (If I recall, there's already an open issue for i18n related macro functions, isn't there?)

I did run into one issue, token size/footprint. _Technically,_ these fall outside of this issue because those values are not stored as an enum, but they do have canonical English names. I'm also kind of afraid of touching the values returned by `getSize()` because I'd imagine that might be more widely used. Should that just stay as is?  (The alternative being to... return them in CAPS or something.)

**Comment by Azhrei** [comment]
My first impression would be to leave them as-is.

They are defined by an XML document and could theoretically be localized there by adding a separate field (one field that defines the size and a new one to define its localized name, with the file name changing to include the locale tacked into the end).

But we know that the sizing system needs a revamp, so effort spent here may be wasted unless that effort is going to include some UI for allowing the GM to define their own sizes (using some kind of visual templating system?).

I’d be inclined to leave it alone for now.  If it becomes more important, it can be revisited in the future.

**Comment by Irarara** [comment]
Good to know! Localizing them for the UI isn't a big deal, yeah. I have something that works.

To me it feels worthwhile to localize them for now if there's no timeline for when the system is going to be changed, but maybe there's no point worrying about the macro functions at least.

**Comment by Phergus** [comment]
Documentation updated for `setTokenVBL()`, `getTokenShape()`, `setTokenShape()` and `getInfo()`.
